pallets/author-reward-dest/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+//! Pallet for setting and getting author reward destination.
+//!
+//! The author of a block can provide an account using inherents, this account is meant to be used
+//! for reward within the block such as tips and part of transaction fees.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use primitives_author_reward_dest::{InherentError, InherentType, INHERENT_IDENTIFIER};
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The type `Self::AccountId` but with additional constraint.
+		type AccountIdType: IsType<<Self as frame_system::Config>::AccountId> + From<InherentType>;
+	}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_finalize(_: BlockNumberFor<T>) {
+			// ensure we never go to trie with these values.
+			<Author<T>>::kill();
```

Comment:
```suggestion
			<Author<T>>::take().expect("Inherent must be present");
```
The `fn is_inherent_required` is a bit weird, see https://github.com/paritytech/polkadot-sdk/issues/3662.  
If you want to ensure that data is present then best to panic `on_finalize`.

Reply:
I definitely agree,
thanks